lib/logger/lib/logger.ex

```diff
@@ -25,23 +25,24 @@ defmodule Logger do
   system. For example, it may be helpful to log whenever a user is deleted.
 
       def delete_user(user) do
-        Logger.info fn ->
-          "Deleting user from the system: #{inspect(user)}"
-        end
+        Logger.info "Deleting user from the system: #{inspect(user)}"
         # ...
       end
 
   The `Logger.info/2` macro emits the provided message at the `:info`
-  level. There are additional macros for other levels. Notice the argument
-  passed to `Logger.info/2` in the above example is a zero argument function.
+  level. Note the arguments given to `info/2` will only be evaluated
```

Comment:
I think if-clauses in first conditional can only use _will_ when they express willingness, as in requests. Then:

“Note the arguments given to `info/2` will only be evaluated if a message is logged.”